x-pack/plugins/infra/public/components/asset_details/lazy_asset_details_wrapper.tsx

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { AssetDetailsProps } from './asset_details';
+
+const AssetDetails = React.lazy(() => import('./asset_details'));
+
+export const LazyAssetDetailsWrapper = (props: AssetDetailsProps) => (
+  <React.Suspense fallback={<div />}>
```

Comment:
I used [log stream embeddable](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/components/log_stream/lazy_log_stream_wrapper.tsx) as an example for the lazy loading but I agree, it's a good idea to have the loading spinner - my only concern is that if something fails to render the embeddable it will show an "infinite" spinner (not sure is that's a real case though) 
I see different lazy loading implementations - also including loading spinner so I guess should be fine